client/landing/gutenboarding/path.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { generatePath, useRouteMatch } from 'react-router-dom';
+import { getLanguageSlugs } from '../../lib/i18n-utils';
+import { ValuesType } from 'utility-types';
+
+export const Step = {
+	IntentGathering: 'about',
+	DesignSelection: 'design',
+	PageSelection: 'pages',
+	Signup: 'signup',
+	CreateSite: 'create-site',
+} as const;
+
+export const langs: string[] = getLanguageSlugs();
+export const steps = Object.keys( Step ).map( key => Step[ key as keyof typeof Step ] );
+
+export const path = `/:step(${ steps.join( '|' ) })/:lang(${ langs.join( '|' ) })?`;
+
+export type StepType = ValuesType< typeof Step >;
+
+export function usePath( step: StepType | undefined, lang?: string ) {
```

Comment:
Should we accept an `undefined` path? That seems like a source of bugs…

```suggestion
export function usePath( step: StepType , lang?: string ) {
```

Reply:
It was added because of `<Link>` accepting also `undefined` and `usePath()` can't be inside conditional. The related lines are: https://github.com/Automattic/wp-calypso/pull/39312/files#diff-5a85a4a3d01d8eee91208a69bb94bd83R53-R54

Reply:
I commented on the change to `Link` [here](https://github.com/Automattic/wp-calypso/pull/39312/files#r380180475).

Reply:
Follow-up here https://github.com/Automattic/wp-calypso/pull/39312#issuecomment-587359480 and https://github.com/Automattic/wp-calypso/pull/39312#discussion_r380547048. Resolving this conversation.